modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/networkapi/model/options.scala

```diff
@@ -0,0 +1,86 @@
+package org.tessellation.rosetta.domain.networkapi.model
+
+import cats.syntax.option._
+
+import org.tessellation.ext.derevo.magnoliaCustomizable.snakeCaseConfiguration
+import org.tessellation.rosetta.domain.currency.Currency
+import org.tessellation.rosetta.domain.error.{RosettaError, RosettaErrorJson}
+import org.tessellation.rosetta.domain.operation.OperationStatus
+import org.tessellation.rosetta.domain.operation.OperationType.Transfer
+import org.tessellation.schema.address.Address
+
+import derevo.cats.{eqv, show}
+import derevo.circe.magnolia.customizableEncoder
+import derevo.derive
+import enumeratum.values.{StringCirceEnum, StringEnum, StringEnumEntry}
+
+object options {
+  @derive(customizableEncoder, eqv, show)
+  case class Version(rosettaVersion: String, nodeVersion: String)
+
+  @derive(customizableEncoder, eqv, show)
+  case class AllowedOperationStatus(status: OperationStatus, successful: Boolean)
+  object AllowedOperationStatus {
+    def apply(opStatus: OperationStatus): AllowedOperationStatus =
+      AllowedOperationStatus(opStatus, OperationStatus.isSuccessful(opStatus))
+
+    def allStatuses: List[AllowedOperationStatus] =
+      OperationStatus.values.map(AllowedOperationStatus(_)).toList
+  }
+
+  @derive(eqv, show)
+  sealed abstract class Case(val value: String) extends StringEnumEntry
+
+  object Case extends StringEnum[Case] with StringCirceEnum[Case] {
+    val values = findValues
+
+    case object CaseSensitive extends Case(value = "case_sensitive")
+  }
+
+  // TODO do we need ExceptionType? If so, which values?
+  @derive(eqv, show)
+  sealed abstract class ExemptionType(val value: String) extends StringEnumEntry
+
+  object ExemptionType extends StringEnum[ExemptionType] with StringCirceEnum[ExemptionType] {
+    val values = findValues
+
+    case object GreaterOrEqual extends ExemptionType(value = "greater_or_equal")
+    case object LessOrEqual extends ExemptionType(value = "less_or_equal")
+    case object Dynamic extends ExemptionType(value = "dynamic")
+  }
+
+  @derive(customizableEncoder, eqv, show)
+  case class BalanceExemption(subAccountAddress: Address, currency: Currency, exemptionType: ExemptionType)
+
+  @derive(customizableEncoder, eqv, show)
+  case class Allow(
+    operationStatuses: List[AllowedOperationStatus],
+    operationTypes: List[String],
+    errors: List[RosettaErrorJson],
+    historicalBalanceLookup: Boolean,
+    timestampStartIndex: Option[Long],
+    callMethods: List[String],
+    balanceExemptions: List[BalanceExemption],
+    mempoolCoins: Boolean,
+    blockHashCase: Option[Case],
+    transactionHashCase: Option[Case]
+  )
+
+  object Allow {
+    val default: Allow = Allow(
+      operationStatuses = AllowedOperationStatus.allStatuses,
+      operationTypes = List(Transfer.toString),
+      errors = RosettaError.values.map(RosettaErrorJson(_)).toList,
+      historicalBalanceLookup = false,
+      timestampStartIndex = none, // TODO will we support this in future?
```

Comment:
We will need to add this in the future, for now it remains none, maybe make a note that this should be a hardcoded constant in the future.